test/reduce-e2e/reduce_test.go

```diff
@@ -33,6 +34,8 @@ type ReduceSuite struct {
 
 // one reduce vertex (keyed)
 func (r *ReduceSuite) TestSimpleKeyedReducePipeline() {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
```

Comment:
Why 5 minutes? Please consider adding a comment on this magic number. It seems shared among all reduce test cases so consider making it a global const.

Reply:
I add timeouts for all the tests I  write. I expect the tests to pass before the timeout.

Reply:
We already set 10 minutes for all e2e tests. https://github.com/numaproj/numaflow/blob/main/Makefile#L115

Reply:
I am ok with keeping the 5 mins timeout. (see comments below)